Sming/system/include/debug_progmem.h

```diff
@@ -63,17 +57,16 @@ extern "C" {
 //Next special version of printf from FakePgmSpace is called to fetch and print the message
 #if DEBUG_PRINT_FILENAME_AND_LINE
 #define debug_e(fmt, ...) \
-	({static const char TOKEN_PASTE2(log_,CUST_FILE_BASE,__LINE__)[] \
+	({static const char log_string[] \
 	__attribute__((aligned(4))) \
-	__attribute__((section(MACROQUOTE(MACROCONCAT(.irom.debug.,__COUNTER__,CUST_FILE_BASE))))) = "[" GET_FNAME2(CUST_FILE_BASE) ":%d] " fmt "\r\n"; \
-	printf_P_stack(TOKEN_PASTE2(log_,CUST_FILE_BASE,__LINE__), __LINE__, ##__VA_ARGS__);})
+	__attribute__((section(MACROQUOTE(MACROCONCAT(.irom.debug.,__COUNTER__,__LINE__))))) = "[" MACROQUOTE(CUST_FILE_BASE) ":%d] " fmt; \
```

Comment:
Please, change
```
= "[" MACROQUOTE(CUST_FILE_BASE) ":%d] " fmt; \
```
to
```
= "[" MACROQUOTE(CUST_FILE_BASE) ":%d] " fmt "\n"; \
```
And remove the new lines from debugf() in Basic_Serial. The change with the new debugf behaviour can be suggested as separate PR. 

The rest of the changes seem to be ok and fix the issue with  non-alpha characters in file name.